File: Selenium/Activities/Activity14.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import java.util.List;

public class Activity14 {
    public static void main(String[] args) {
        WebDriverManager.firefoxdriver().setup();

        // Suppressing browser log
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "NULL");
        // Initialize firefox driver
        WebDriver driver = new FirefoxDriver();

        // Open a page
        driver.get("https://training-support.net/selenium/tables");

        //Page title
        System.out.println("Title of page is: " + driver.getTitle());

        // Listing rows and column count
        List<WebElement> rows = driver.findElements(By.xpath("//table[@id='sortableTable']/tbody/tr"));
        List<WebElement> cols = driver.findElements(By.xpath("//table[@id='sortableTable']/thead/tr/th"));
        System.out.println("Number of rows: " + rows.size());
        System.out.println("Number of cols: " + cols.size());

        //cell value at the second row second column
        WebElement row2col2 = driver.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]"));
        System.out.println(row2col2.getText());

        driver.findElement(By.xpath("//table[@id='sortableTable']/thead/tr/th[1]")).click();

        row2col2 = driver.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]"));
        System.out.println(row2col2.getText());

        // Print the cell values of the footer
        List<WebElement> footer = driver.findElements(By.xpath("//table[@id='sortableTable']/tfoot/tr/th"));
        System.out.println("Cell values of the footer: ");
        for(WebElement cell : footer) {
            System.out.println(cell.getText());
        }

        driver.quit();
    }
}

File: Selenium/Activities/Activity14.py
from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.common.by import By

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Open the link in browser
    driver.get("https://training-support.net/selenium/tables")

    # Print page title
    print("Page Title is: " + driver.title)

    # Listing rows and column count
    rows = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/tbody/tr")
    cols = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/thead/tr/th")
    print(len(rows))
    print(len(cols))

    # cell value at the second row second column
    row2col2 = driver.find_element(By.XPATH, "//table[@id='sortableTable']/tbody/tr[2]/td[2]")
    print(row2col2.text)

    driver.find_element(By.XPATH, "//table[@id='sortableTable']/thead/tr/th[1]").click()

    row2col2 = driver.find_element(By.XPATH, "//table[@id='sortableTable']/tbody/tr[2]/td[2]")
    print(row2col2.text)

    # Print the cell values of the footer
    footer = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/tfoot/tr/th")
    print("Cell values of the footer: ");
    for cell in footer:
        print(cell.text);

    driver.quit()

